src/mp3/read.rs

```diff
@@ -80,17 +80,30 @@ where
 					continue;
 				}
 			},
-			_ if verify_frame_sync([header[0], header[1]]) => {
-				let start = reader.seek(SeekFrom::Current(0))? - 4;
-				let header = Header::read(u32::from_be_bytes(header))?;
-
-				file.first_frame_offset = Some(start);
-				first_frame_header = Some(header);
-
-				// We have found the first frame
-				break;
+			// metadata blocks might be followed by junk bytes before the first MP3 frame begins
+			_ => {
+				// seek back the length of the temporary header buffer
+				// so that all bytes are included in the search for a frame sync
+				let start_of_search_area =
+					reader.seek(SeekFrom::Current(-1 * header.len() as i64))?;
+				if let Some(first_mp3_frame_start_relative) = search_for_frame_sync(reader)? {
+					let first_mp3_frame_start_absolute =
+						start_of_search_area + first_mp3_frame_start_relative;
+
+					// read the first four bytes of the found frame
+					reader.seek(SeekFrom::Start(first_mp3_frame_start_absolute))?;
```

Comment:
Aren't we already here?

Reply:
No, because the call to `search_for_frame_sync(reader)` changes the reader and the first two bytes of the MP3 frame are consumed. The function needs to read the first 11 bits, i.e. the first two bytes, and those would not be included in the follwing `reader.read_u32`.

It could be replaced by `reader.Seek(SeekFrom::Current(-2))`, but I think this would be less readable and robust.

Reply:
Ah, yeah this is definitely the more readable option. Forgot this will leave the reader past the frame sync.